Update docs/heading icon page content (#2238)
vanilla-framework/vanilla-framework: docs/en/patterns/heading-icon.md

## Heading icon

<hr>

> A header can be formatted to emphasise an icon and supplementary text.

Reviewer: ```suggestion
A header can be emphasised by adding an icon alongside the text.
```

<a href="https://vanilla-framework.github.io/vanilla-framework/examples/patterns/heading-icon/"
  class="js-example">
View example of the pattern heading icons
</a>

### Stacked

> Available as a vertical variant allowing for an alternate layout.

Reviewer: ```suggestion
This variant positions the icon vertically with the text content for an alternate layout.
```

<a href="https://vanilla-framework.github.io/vanilla-framework/examples/patterns/heading-icon-stacked/"
  class="js-example">
View example of the pattern heading icon stacked
</a>

### Small

> Available at a smaller size of 32 x 32 pixels rather than our default size of 60 x 60 pixels.

Reviewer: ```suggestion
The icon for this pattern is also available at a smaller size of 32 x 32 pixels rather than our default size of 60 x 60 pixels.
```
